citation-server/src/graphql/query.js

```diff
@@ -20,71 +21,61 @@ export const ObjectInterface = new GraphQLInterfaceType({
 	})
 });
 
-export const PageType = new GraphQLObjectType({
-	name: 'Page',
-	interfaces: [ObjectInterface],
-	isTypeOf: value => value.__type__ === 'Page',
-	fields: () => ({
-		__id__: {type: GraphQLID},
-		__type__: {type: GraphQLString},
-		__tree__: {
-			type: GraphQLString,
-			resolve: inspect
-		},
-		__newId__: {type: GraphQLID},
-		slug: {type: GraphQLString},
-		title: {type: GraphQLString},
-		children: {
-			type: new GraphQLList(PageType),
-			resolve: root => readChildren(root.children)
-		},
-		component: {
-			type: ComponentType,
-			resolve: root => readChild(root.component)
-		}
-	})
-});
-
-export const ComponentType = new GraphQLObjectType({
-	name: 'Component',
-	interfaces: [ObjectInterface],
-	isTypeOf: value => value.__type__ === 'Component',
-	fields: () => ({
-		__id__: {type: GraphQLID},
-		__type__: {type: GraphQLString},
-		__tree__: {
-			type: GraphQLString,
-			resolve: inspect
-		},
-		__newId__: {type: GraphQLID},
-		type: {type: GraphQLString},
-		children: {
-			type: new GraphQLList(ComponentType),
-			resolve: root => readChildren(root.children)
-		},
-		data: {
-			type: new GraphQLList(ObjectInterface),
-			resolve: root => readChildren(root.data)
-		}
-	})
-});
-
-export const ContentType = new GraphQLObjectType({
-	name: 'Content',
-	interfaces: [ObjectInterface],
-	isTypeOf: value => value.__type__ === 'Content',
-	fields: () => ({
-		__id__: {type: GraphQLID},
-		__type__: {type: GraphQLString},
-		__tree__: {
-			type: GraphQLString,
-			resolve: inspect
-		},
-		__newId__: {type: GraphQLID},
-		title: {type: GraphQLString},
-		content: {type: GraphQLString}
-	})
-});
+export async function buildObjects() {
+	const ObjectTypes = {};
+	const model = await readModel();
+	for (const structure of model) {
+		ObjectTypes[structure.name] = new GraphQLObjectType({
+			name: structure.name,
+			interfaces: [ObjectInterface],
+			isTypeOf: value => value.__type__ === structure.name,
+			fields: () => {
+				const resultFields = {
+					__id__: {type: GraphQLID},
+					__newId__: {type: GraphQLID},
+					__type__: {type: GraphQLString},
+					__tree__: {
+						type: GraphQLString,
+						resolve: inspect
+					}
+				};
+				structure.fields.forEach(field => {
+					if (typeof field.type === 'string') {
+						resultFields[field.name] = {type: GraphQLString};
+					} else if (typeof field.type === 'object') {
```

Comment:
Array.isArray or perhaps _.isString & _.isArray would be better than manual typeof